### ISSUE TITLE
Order managers should be able to print invoices / pick sheets in bulk

### ISSUE THREAD

**Issue by spencern** [reactioncommerce/reaction#2343]
Users with permission to view and process orders should be able to print invoices for all selected orders.

These invoices should be consecutively displayed ready for print in a single document with page breaks between each order.

Once the print document is generated, the orders should all have an "Invoice Printed" event added to their order history. (There can, and should, be multiple invoice printed events if the invoice is printed multiple times)

Any orders that have a "New" status or status that comes before "Printed" should be progressed to the "Printed" state. Any orders that have a more advanced status should not be regressed.

See: #2325 for how to select orders for bulk processing.

**Comment by rymorgan** [comment]
**Links for designs below** 

Reaction core team link: 
https://zpl.io/Z1MCuWh

Public link: 
https://scene.zeplin.io/project/58af8900737c137280322bd6/screen/58faa0e175ba8a9495284660

**Comment by kieha** [comment]
@spencern @rymorgan a couple of questions on this:
- Printing a pick sheet should assume that you've already picked the order/items?
- What should/does a pick sheet look like? Should there be designs for that?

**Comment by rymorgan** [comment]
> Printing a pick sheet should assume that you've already picked the order/items?

@kieha - There needs to be a step before picking that says print ( @spencern pointed this out to me when we were working on split fulfillment) I'll make that adjustment here as well. So, in other words, it should start with print, then pick... 

> What should/does a pick sheet look like? Should there be designs for that?
As far as the pick sheet, it should probably have some simple design but I also need to know from @spencern what those requirements would be. 

> In the sidebar flow, one can print an invoice before capturing the payment, or after capturing the payment, but only after approving the payment. Should this be the same kind of flow for the bulk actions bar? If so, there need to be more designs for this section, IMHO.

I'm not entirely convinced the approve step is necessary. Why do you need to approve and then capture, what is approving doing? I realize this isn't actionable, but I think it deserves a conversation.

**Comment by zenweasel** [comment]
Warning: Unsolicited input:

It would be cool if it was configurable (i.e. you could turn it off) but one way that approve can be used would be for fraud screening, especially when we get the "risk assessment" feature implemented. You don't want fraud orders flowing into your system and allocating inventory before canceling them.

**Comment by spencern** [comment]
@rymorgan 

In regards to approve vs Capture
I'm also not sure what the difference is between those two steps. Perhaps some payment providers require approval of charge before it's captured or authorized? 

@brent do you know anything about this approval step?

**Comment by spencern** [comment]
@zenweasel I agree that an order triage feature would be a quite helpful, a great idea. 

Perhaps another queue for certain user types/groups that permits where elevated or high risk transactions go before getting approved and moved into the standard order queue, or cancelled.

**Comment by rymorgan** [comment]
I'm fairly certain that the approval step was added by @mikemurray and isn't a requirement. I'm proposing removing it unless it's required. 

I really like the order triage idea.

**Comment by kieha** [comment]
@rymorgan @spencern alright, I'll skip over the `approve` step for now.

Another question I'd bring up from this would be, if it's possible to print an invoice either before or after capturing a payment, should the `Print Invoice` button (or whatever it's equivalent) be an extra button on the bar (in the same way `Capture` is set aside), or should it be in the list under the `Bulk Actions` section?

**Comment by spencern** [comment]
I'll defer to @rymorgan on that.

**Comment by rymorgan** [comment]
This should be solved with a print step where we print both an invoice and pick sheet. I'll add that to flow and update the zeplin files. The flow will be: Print Invoice & Pick Sheet > Pick Items/Shipment/Order > Pack Items/Shipment/Order > Label Shipment > Ship

**Comment by rymorgan** [comment]
I put together a quick prototype of this flow: https://invis.io/S5CK720KW

Click shipped at the end to run through it again.  
![order-fulfillment](https://user-images.githubusercontent.com/20409254/29436296-b344e542-835f-11e7-8f5b-00cef06b9375.gif)

**Comment by kieha** [comment]
@rymorgan any progress on the pick sheets design? As for the invoice, should I just use the one that's already there for individual order invoice or did you have something else in mind?

The existing invoice design: 

<img width="1588" alt="screen shot 2017-08-25 at 7 41 31 pm" src="https://user-images.githubusercontent.com/19430095/29723484-77ee6dc8-89cd-11e7-9710-c262602730fb.png">

**Comment by rymorgan** [comment]
@kieha I haven't been able to get to it yet. You can use the current invoice for now. I plan on cleaning up the design for that and pick sheet, but I doubt I'll get to it today. So if you can make any sort of rough version of that then we can move forward that way and implement my design later. (hopefully early next week.)
